pandas/io/common.py

```diff
@@ -324,9 +325,51 @@ def _get_handle(path, mode, encoding=None, compression=None):
         else:
             f = open(path, mode)
 
+    if memory_map and hasattr(f, 'fileno'):
+        try:
+            f = MMapWrapper(f)
+        except:  # fallback: leave file handler as is
```

Comment:
Also, non-existent files won't show up when you instantiate `MMapWrapper`, as the constructor is expecting a `file` object, which should already be pointing to an existing or newly-created file.